app/modules/asset/service.server.ts

```diff
@@ -671,13 +672,11 @@ export const getPaginatedAndFilterableAssets = async ({
   const cookie = await updateCookieWithPerPage(request, perPageParam);
   const { perPage } = cookie;
 
-  const categories = await getAllCategories({
-    userId,
-  });
+  const categories = await db.category.findMany({ where: { userId }, take: 4 });
```

Comment:
Here you have 4 queries that will run one after the other. Please put them in a transaction like the example above.